packages/babel-traverse/src/scope/index.js

```diff
@@ -140,7 +140,7 @@ const collectorVisitor = {
   },
 
   UpdateExpression(path, state) {
-    state.constantViolations.push(path.get("argument"));
+    state.constantViolations.push(path);
   },
 
   UnaryExpression(path, state) {
```

Comment:
Because I don't understand all the implications of delete in sloppy code, so best to leave this in if it's already here. And, we need to make it consistent with the other changes you've made.